src/containers/collaborations/components/AddDataSetToCollaborationModal.js

```diff
@@ -0,0 +1,193 @@
+/*
+ * @flow
+ */
+
+import React, { useEffect, useState } from 'react';
+
+import { List, Map } from 'immutable';
+import { Modal, Select, Typography } from 'lattice-ui-kit';
+import { ReduxUtils, useRequestState, useStepState } from 'lattice-utils';
+import { useDispatch, useSelector } from 'react-redux';
+import type { Organization, UUID } from 'lattice';
+import type { RequestState } from 'redux-reqseq';
+
+import SearchOrgDataSetsContainer from '../../org/components/dataset/SearchOrgDataSetsContainer';
+import StyledFooter from '../../org/components/styled/StyledFooter';
+import { ModalBody, ResetOnUnmount, StackGrid } from '../../../components';
+import { COLLABORATIONS } from '../../../core/redux/constants';
+import { selectOrganizations } from '../../../core/redux/selectors';
+import { ADD_DATA_SETS_TO_COLLABORATION, addDataSetsToCollaboration } from '../actions';
+import type { ReactSelectOption } from '../../../types';
+
+const { isPending, isStandby, isSuccess } = ReduxUtils;
+
+const RESET_ACTIONS = [ADD_DATA_SETS_TO_COLLABORATION];
+
+const AddDataSetToCollaborationModalBody = ({
+  collaborationId,
+  existingDataSets,
+  isVisible,
+  onClose,
+  organizationIds,
+  ...rest
+} :{
+  collaborationId :UUID;
+  existingDataSets :List;
+  isVisible :boolean;
+  onClose :() => void;
+  organizationIds :List<UUID>;
+}) => {
+  const dispatch = useDispatch();
+  const [modalTitle, setModalTitle] = useState(<Typography variant="h3">Select Organization</Typography>);
+  const [step, stepBack, stepNext, setStep] = useStepState(3);
+  const [selectedOrganization, setSelectedOrganization] = useState('');
+  const [selectedDataSets, setSelectedDataSets] = useState(List());
+  const [selectedDataSetTitles, setSelectedDataSetTitles] = useState(List());
+  const addDataSetToCollaborationRS :?RequestState = useRequestState([COLLABORATIONS, ADD_DATA_SETS_TO_COLLABORATION]);
+
+  const organizations :Map<UUID, Organization> = useSelector(selectOrganizations());
+
+  const pending = isPending(addDataSetToCollaborationRS);
+  const standby = isStandby(addDataSetToCollaborationRS);
+  const success = isSuccess(addDataSetToCollaborationRS);
+
+  const orgOptions = organizationIds.filter((orgId) => organizations.has(orgId))
+    .map((orgId) => {
+      const organization = organizations.get(orgId);
+      const label = organization.title;
+      const value = organization.id;
+      return { label, value };
+    }).toJS();
+
+  useEffect(() => {
+    if (step === 0) {
+      setModalTitle(<Typography variant="h3">Select Organization</Typography>);
+    }
+    else if (step === 1) {
+      setModalTitle(<Typography variant="h3">Add Data Sets</Typography>);
+    }
+    else if (standby && step === 2) {
+      setModalTitle(<Typography variant="h3">Add these data sets?</Typography>);
+    }
+    else if (success && step === 2) {
+      setModalTitle(<Typography variant="h3">Data Sets Added</Typography>);
+    }
```

Comment:
You don't need to save the entire component in state, you can just save the title string.